pkg/drift_client/package.py

```diff
@@ -0,0 +1,120 @@
+"""Wrapper around DriftPackage"""
+from typing import Optional
+
+from wavelet_buffer import WaveletBuffer  # pylint: disable=no-name-in-module
+from drift_protocol.meta import MetaInfo
+from drift_protocol.common import DataPayload, DriftPackage, StatusCode
+import numpy as np
+
+
+def check_status(func):
+    """Check Package status"""
+
+    def dec(self):
+        if self._pkg.status != StatusCode.GOOD:  # pylint: disable=protected-access
+            raise ValueError("Bad package")
+        return func(self)
+
+    return dec
+
+
+class Package:  # pylint: disable=no-member
+    """Parsed Drift Package Class"""
+
+    _pkg: DriftPackage
+
+    def __init__(self, blob: bytes):
+        """Parsed Drift Package
+
+        Args:
+            blob: Serialized  package from database or stream
+        """
+        pkg = DriftPackage()
+        pkg.ParseFromString(blob)
+        self._pkg = pkg
+
+    TS_PRECISION = 1000
+
+    @property
+    def package_id(self) -> int:
+        """Package ID
+
+        Returns:
+            Package ID (by this all acquired data is synced)
+        """
+        return self._pkg.id
+
+    @property
+    def source_timestamp(self) -> float:
+        """Source Timestamp
+
+        Returns:
+            Source timestamp (Timestamp when the service
+                has received  the input package)
+        """
+        return self._pkg.source_timestamp.ToMilliseconds() / self.TS_PRECISION
+
+    @property
+    def publish_timestamp(self) -> float:
+        """Publish Timestamp
+
+        Returns:
+            Publish timestamp (Timestamp when the service
+                has done its job and sends the output package.)
+        """
+        return self._pkg.publish_timestamp.ToMilliseconds() / self.TS_PRECISION
+
+    @property
+    def status_code(self) -> int:
+        """Status Code
+
+        Returns:
+            Status of the package. Ok (0) means the package is valid
+        """
+        return self._pkg.status
```

Comment:
Isn't there some Enum type for this?

Reply:
Currently. a user can use protobuf StatusCode enum.